azurerm_data_protection_backup_policy_postgresql - criteria block is required, but provider crashes if given an empty object

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.2.0

### AzureRM Provider Version

3.16.0

### Affected Resource(s)/Data Source(s)

azurerm_data_protection_backup_policy_postgresql

### Terraform Configuration Files

```hcl
Defining a retention rule without criteria such as:

retention_rule {
   duration = "P31D"
   name     = "postgresql_daily"
   priority = 40

   criteria {}
}

Will cause a provider panic.  The docs say this is a required field.  Setting a value inside of this criteria fixes the problem.
```


### Debug Output/Panic Output

```shell
Error: Plugin did not respond


Stack trace from the terraform-provider-azurerm_v3.16.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 122 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/dataprotection.expandBackupPolicyPostgreSQLCriteriaArray({0xc002cccbe0, 0x1, 0x60b8575?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/dataprotection/data_protection_backup_policy_postgresql_resource.go:384 +0xbd4
github.com/hashicorp/terraform-provider-azurerm/internal/services/dataprotection.expandBackupPolicyPostgreSQLTaggingCriteriaArray({0xc002cb0900, 0x4, 0x68265e0?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/dataprotection/data_protection_backup_policy_postgresql_resource.go:369 +0x259
github.com/hashicorp/terraform-provider-azurerm/internal/services/dataprotection.resourceDataProtectionBackupPolicyPostgreSQLCreate(0x0?, {0x572b400?, 0xc00240a380})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/dataprotection/data_protection_backup_policy_postgresql_resource.go:203 +0x53c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x68303a8?, {0x68303a8?, 0xc003455ef0?}, 0xd?, {0x572b400?, 0xc00240a380?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0005a2460, {0x68303a8, 0xc003455ef0}, 0xc002ffdba0, 0xc003445e80, {0x572b400, 0xc00240a380})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:837 +0xa7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc001588ca8, {0x68303a8?, 0xc003455e30?}, 0xc003452820)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0016c2500, {0x68303a8?, 0xc0034558c0?}, 0xc0000e97a0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:813 +0x4fc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5cecee0?, 0xc0016c2500}, {0x68303a8, 0xc0034558c0}, 0xc002fefb60, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000226a80, {0x6841c08, 0xc0012d4b60}, 0xc00345a900, 0xc0017526f0, 0xa87c780, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc000226a80, {0x6841c08, 0xc0012d4b60}, 0xc00345a900, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1620 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.47.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.47.0/server.go:920 +0x28a

Error: The terraform-provider-azurerm_v3.16.0_x5 plugin crashed!
```


### Expected Behaviour

A validation error should be thrown indicating that criteria is required.

### Actual Behaviour

A panic and obscure "Error: Plugin did not respond" error.

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/dataprotection/data_protection_backup_policy_postgresql_resource.go
+++ b/internal/services/dataprotection/data_protection_backup_policy_postgresql_resource.go
@@ -200,7 +200,10 @@ func resourceDataProtectionBackupPolicyPostgreSQLCreate(d *pluginsdk.ResourceDat
 		return tf.ImportAsExistsError("azurerm_data_protection_backup_policy_postgresql", id.ID())
 	}
 
-	taggingCriteria := expandBackupPolicyPostgreSQLTaggingCriteriaArray(d.Get("retention_rule").([]interface{}))
+	taggingCriteria, err := expandBackupPolicyPostgreSQLTaggingCriteriaArray(d.Get("retention_rule").([]interface{}))
+	if err != nil {
+		return err
+	}
 	policyRules := make([]dataprotection.BasicBasePolicyRule, 0)
 	policyRules = append(policyRules, expandBackupPolicyPostgreSQLAzureBackupRuleArray(d.Get("backup_repeating_time_intervals").([]interface{}), taggingCriteria)...)
 	policyRules = append(policyRules, expandBackupPolicyPostgreSQLDefaultAzureRetentionRule(d.Get("default_retention_duration")))
@@ -351,7 +354,7 @@ func expandBackupPolicyPostgreSQLDefaultAzureRetentionRule(input interface{}) da
 	}
 }
 
-func expandBackupPolicyPostgreSQLTaggingCriteriaArray(input []interface{}) *[]dataprotection.TaggingCriteria {
+func expandBackupPolicyPostgreSQLTaggingCriteriaArray(input []interface{}) (*[]dataprotection.TaggingCriteria, error) {
 	results := []dataprotection.TaggingCriteria{
 		{
 			Criteria:        nil,
@@ -365,20 +368,29 @@ func expandBackupPolicyPostgreSQLTaggingCriteriaArray(input []interface{}) *[]da
 	}
 	for _, item := range input {
 		v := item.(map[string]interface{})
-		results = append(results, dataprotection.TaggingCriteria{
-			Criteria:        expandBackupPolicyPostgreSQLCriteriaArray(v["criteria"].([]interface{})),
+		result := dataprotection.TaggingCriteria{
 			IsDefault:       utils.Bool(false),
 			TaggingPriority: utils.Int64(int64(v["priority"].(int))),
 			TagInfo: &dataprotection.RetentionTag{
 				ID:      utils.String(v["name"].(string) + "_"),
 				TagName: utils.String(v["name"].(string)),
 			},
-		})
+		}
+		criteria, err := expandBackupPolicyPostgreSQLCriteriaArray(v["criteria"].([]interface{}))
+		if err != nil {
+			return nil, err
+		}
+		result.Criteria = criteria
+
+		results = append(results, result)
 	}
-	return &results
+	return &results, nil
 }
 
-func expandBackupPolicyPostgreSQLCriteriaArray(input []interface{}) *[]dataprotection.BasicBackupCriteria {
+func expandBackupPolicyPostgreSQLCriteriaArray(input []interface{}) (*[]dataprotection.BasicBackupCriteria, error) {
+	if len(input) == 0 || input[0] == nil {
+		return nil, fmt.Errorf("criteria is a required field, cannot leave blank")
+	}
 	results := make([]dataprotection.BasicBackupCriteria, 0)
 	for _, item := range input {
 		v := item.(map[string]interface{})
@@ -429,7 +441,7 @@ func expandBackupPolicyPostgreSQLCriteriaArray(input []interface{}) *[]dataprote
 			ObjectType:       dataprotection.ObjectTypeBasicBackupCriteriaObjectTypeScheduleBasedBackupCriteria,
 		})
 	}
-	return &results
+	return &results, nil
 }
 
 func flattenBackupPolicyPostgreSQLBackupRuleArray(input *[]dataprotection.BasicBasePolicyRule) []interface{} {

